jina/serve/runtimes/gateway/request_handling.py

```diff
@@ -157,18 +159,23 @@ async def _load_balance(self, request):
         try:
             async with aiohttp.ClientSession() as session:
 
-                if request.method == 'GET':
-                    request_kwargs = {}
-                    try:
-                        payload = await request.json()
-                        if payload:
-                            request_kwargs['json'] = payload
-                    except Exception:
-                        self.logger.debug('No JSON payload found in request')
-
-                    async with session.get(
-                        url=target_url, **request_kwargs
-                    ) as response:
+                request_kwargs = {
+                    'headers': request.headers,
+                    'params': request.query,
+                }
+                try:
+                    payload = await request.content.read()
+                    if payload:
+                        request_kwargs['data'] = payload
+                except Exception:
+                    self.logger.debug('No JSON payload found in request')
+
+                async with _RequestContextManager(
+                    session._request(request.method, target_url, **request_kwargs)
+                ) as response:
```

Comment:
@NarekA why not directly use the `request` method from the session?, it's already wrapping the context manager for you.

```
    def request(
        self, method: str, url: StrOrURL, **kwargs: Any
    ) -> "_RequestContextManager":
        """Perform HTTP request."""
        return _RequestContextManager(self._request(method, url, **kwargs))
        ```

Reply:
Somehow I missed this, will fix.